packages/medusa/src/models/batch-job.ts

```diff
@@ -22,15 +25,31 @@ export class BatchJob {
   @DbAwareColumn({ type: "enum", enum: BatchJobStatus })
   status: BatchJobStatus
 
-  @Column({ type: "text", nullable: true })
+  @Column({ nullable: true })
   created_by: string | null
 
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "created_by" })
+  created_by_user: User
+
   @DbAwareColumn({ type: "jsonb", nullable: true })
   context: Record<string, unknown>
 
   @DbAwareColumn({ type: "jsonb", nullable: true })
   result: Record<string, unknown>
 
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  processing_at: Date | null
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  awaiting_confirmation_at: Date | null
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  confirmed_at: Date | null
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  cancelled_at: Date | null
+
```

Comment:
we have a `confirmed_at` column, but I have added a `completed_at` as well to be set after completing (assuming that confirming will be set and the complete process begins)

Reply:
Exactly - the idea is that the client will inform the BatchService that we can confirm the job, which will record the timestamp and notify the worker that it can now start processing the job. Once the worker is done it will notify back to the BatchService that the job is "Complete" which will set `completed_at`.